File: c/gpt/p81.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    char *buf = malloc(1024);
    if (!buf) {
        fprintf(stderr, "malloc failed\n");
        return 1;
    }

    printf("Enter input (up to 10 chars will be read):\n");
    /* fgets reads up to n-1 characters; here we pass 10 so it will read up to 9 chars plus NUL */
    if (fgets(buf, 10, stdin) != NULL) {
        printf("Read: %s\n", buf);
    } else {
        printf("No input read.\n");
    }

    free(buf);
    return 0;
}

File: javascript/gpt/p81.js
#!/usr/bin/env node
const readline = require('readline');

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

rl.question('Enter input (up to 10 characters will be read):\n', (answer) => {
  // allocate a 1024-byte buffer
  const buf = Buffer.alloc(1024);
  const slice = Buffer.from(answer, 'utf8').slice(0, 10);
  slice.copy(buf, 0);
  console.log('Read:', slice.toString('utf8'));
  rl.close();
});
